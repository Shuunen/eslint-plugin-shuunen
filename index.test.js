import { expect, it } from 'vitest'

const index = require('./index')

it('index', () => {
  const text = JSON.stringify(index, undefined, 2)
  const result = text.replace(/"version": "[^"]+"/, '"version": "0.0.0"')
  expect(result).toMatchSnapshot()
})
