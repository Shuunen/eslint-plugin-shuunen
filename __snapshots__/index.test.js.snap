// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`index 1`] = `
"{
  "meta": {
    "name": "eslint-plugin-shuunen",
    "version": "0.0.0"
  },
  "rules": {
    "avoid-name": {
      "meta": {
        "schema": [],
        "type": "suggestion",
        "docs": {
          "description": "disallow variables named \`foo\`, it is so old school",
          "recommended": true
        },
        "fixable": "code",
        "messages": {
          "avoidName": "Avoid using variables named '{{ name }}'"
        }
      }
    }
  },
  "configs": {
    "recommended": {
      "env": {
        "es2024": true
      },
      "parserOptions": {
        "ecmaVersion": "latest",
        "sourceType": "module"
      },
      "plugins": [
        "shuunen"
      ],
      "rules": {
        "shuunen/avoid-name": "error"
      }
    },
    "all": {
      "env": {
        "es2024": true
      },
      "parserOptions": {
        "ecmaVersion": "latest",
        "sourceType": "module"
      },
      "plugins": [
        "shuunen"
      ],
      "rules": {
        "shuunen/avoid-name": "error"
      }
    },
    "flat/recommended": {
      "languageOptions": {
        "globals": {
          "ArrayBuffer": false,
          "DataView": false,
          "Float32Array": false,
          "Float64Array": false,
          "Int16Array": false,
          "Int32Array": false,
          "Int8Array": false,
          "Map": false,
          "Promise": false,
          "Proxy": false,
          "Reflect": false,
          "Set": false,
          "Symbol": false,
          "Uint16Array": false,
          "Uint32Array": false,
          "Uint8Array": false,
          "Uint8ClampedArray": false,
          "WeakMap": false,
          "WeakSet": false,
          "Atomics": false,
          "SharedArrayBuffer": false,
          "BigInt": false,
          "BigInt64Array": false,
          "BigUint64Array": false,
          "globalThis": false,
          "AggregateError": false,
          "FinalizationRegistry": false,
          "WeakRef": false
        }
      },
      "name": "shuunen/flat/recommended",
      "plugins": {
        "shuunen": {
          "meta": {
            "name": "eslint-plugin-shuunen",
            "version": "0.0.1"
          },
          "rules": {
            "avoid-name": {
              "meta": {
                "schema": [],
                "type": "suggestion",
                "docs": {
                  "description": "disallow variables named \`foo\`, it is so old school",
                  "recommended": true
                },
                "fixable": "code",
                "messages": {
                  "avoidName": "Avoid using variables named '{{ name }}'"
                }
              }
            }
          }
        }
      },
      "rules": {
        "shuunen/avoid-name": "error"
      }
    },
    "flat/all": {
      "languageOptions": {
        "globals": {
          "ArrayBuffer": false,
          "DataView": false,
          "Float32Array": false,
          "Float64Array": false,
          "Int16Array": false,
          "Int32Array": false,
          "Int8Array": false,
          "Map": false,
          "Promise": false,
          "Proxy": false,
          "Reflect": false,
          "Set": false,
          "Symbol": false,
          "Uint16Array": false,
          "Uint32Array": false,
          "Uint8Array": false,
          "Uint8ClampedArray": false,
          "WeakMap": false,
          "WeakSet": false,
          "Atomics": false,
          "SharedArrayBuffer": false,
          "BigInt": false,
          "BigInt64Array": false,
          "BigUint64Array": false,
          "globalThis": false,
          "AggregateError": false,
          "FinalizationRegistry": false,
          "WeakRef": false
        }
      },
      "name": "shuunen/flat/all",
      "plugins": {
        "shuunen": {
          "meta": {
            "name": "eslint-plugin-shuunen",
            "version": "0.0.1"
          },
          "rules": {
            "avoid-name": {
              "meta": {
                "schema": [],
                "type": "suggestion",
                "docs": {
                  "description": "disallow variables named \`foo\`, it is so old school",
                  "recommended": true
                },
                "fixable": "code",
                "messages": {
                  "avoidName": "Avoid using variables named '{{ name }}'"
                }
              }
            }
          }
        }
      },
      "rules": {
        "shuunen/avoid-name": "error"
      }
    }
  }
}"
`;
